odoo/addons/base/models/ir_actions.py

```diff
@@ -1064,6 +1064,8 @@ def _eval_value(self, eval_context=None):
             elif action.update_field_id.ttype in ['many2one', 'integer']:
                 try:
                     expr = int(action.value)
+                    if expr == 0 and action.update_field_id.ttype == 'many2one':
```

Comment:
You could be more generic.
```suggestion
                    if expr == 0 and action.update_field_id.relational:
```

Reply:
Because of the `if` above, we can only have `many2one` of `integer` fields.  Generalizing is pointless IMHO.